src/applications/vaos/express-care/redux/actions.js

```diff
@@ -145,14 +145,35 @@ export function routeToPageInFlow(flow, history, current, action) {
       type: FORM_PAGE_CHANGE_STARTED,
     });
 
-    const nextAction = flow[current][action];
+    const state = getState();
     let nextPage;
 
-    if (typeof nextAction === 'string') {
-      nextPage = flow[nextAction];
+    if (action === 'next') {
+      const nextAction = flow[current][action];
+
+      if (typeof nextAction === 'string') {
+        nextPage = flow[nextAction];
+      } else {
+        const nextStateKey = await nextAction(getState(), dispatch);
+        if (
+          Object.keys(state.expressCare.newRequest.previousPages).length <= 0
+        ) {
+          Object.assign(state.expressCare.newRequest.previousPages, {
+            home: current,
+          });
+        }
+        Object.assign(state.expressCare.newRequest.previousPages, {
+          [current]: nextStateKey,
```

Comment:
So, interestingly, I think the way you have the key and value here makes `previousPages` more of a `nextPages` object.

This isn't wrong, it just shifts where some of the code is written. Written this way, it makes handling setting `home` a little easier, but it forces you to do a `find` below.

If you flip these around and do `[nextStateKey]: current`, then you have to handle `home` differently, but line 171 below becomes `const previousPage = state.expressCare.newRequest.previousPages[current];`.

Reply:
Understood.

Reply:
Adjust object style